include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,819 @@
+  *
+  * for decompression+hash, call returns with hash appended to the end of
+  * last processed chunk of data.
+  *
+  * User should compute processed data len = total output length - digest_len,
+  * where digest_len queried through odp_comp_hash_alg_capability().
+  *
+  * If operation updates result structure with status
+  * ODP_COMP_ERR_OUT_OF_SPACE then application
+  * should call API again with valid output buffer (and no-more input)
+  * until call completes with any other error code.
+  *
+  * Please note it is always recommended that application using async mode,
+  * provide sufficiently large buffer size to avoid ODP_COMP_ERR_OUT_OF_SPACE.
+  * Else it is recommended that application maintain required context
+  * to associate event to its respective input.
+  *
+  * @param param[in]          Operation parameters.
+  *
+  * @retval 0 on success
+  * @retval <0 on failure
+  */
+int odp_comp_decomp_enq(odp_comp_op_param_t *param);
+
+ /**
+  * Convert processed packet event to packet handle
+  *
+  * Get packet handle corresponding to processed packet event. Event subtype
+  * must be ODP_EVENT_PACKET_COMP. compression/decompression operation
+  * results can be examined with odp_comp_result().
+  *
+  * @param ev	    Event handle
+  *
+  * @return Valid Packet handle on success,
+  *	    ODP_PACKET_INVALID on failure
+  *
+  * @see odp_event_subtype(), odp_comp_result()
+  *
+  * @ Example Usage
+  * odp_event_t ev = odp_queue_deque(comple_q);
+  * odp_event_subtype_t subtype;
+  * if(ODP_PACKET_EVENT == odp_event_types(ev, &subtype)) {
+  * if(subtype == ODP_PACKET_EVENT_COMP) {
+  *  pkt = odp_comp_packet_from_event(ev);
+  *  odp_comp_op_result_t res;
+  *  odp_comp_result(packet, &res);
+  * }
+  * }
+  */
+odp_packet_t odp_comp_packet_from_event(odp_event_t event);
+
+ /**
+  * Get compression/decompression operation results from an processed packet.
+  *
+  * Successful compression/decompression operations produce
+  * packets which contain operation result metadata. This function copies the
+  * operation results from an processed packet. Event subtype of this kind
+  * of packet is ODP_EVENT_PACKET_COMP. Results are undefined if input packet 
+  * has not be processed by compression/decompression call.
+  *
+  * @param[out]    result  Pointer to operation result for output
+  * @param	   packet  An processed packet (ODP_EVENT_PACKET_COMP)
+  *
+  * @retval  0	   On success
+  * @retval <0	   On failure
+  *
+  * @see odp_comp_compress_enq(), odp_comp_decomp_enq(),
+  *	  odp_comp_packet_from_event()
+  */
+int odp_comp_result(odp_packet_t packet,
+		    odp_comp_op_result_t *result);
+
+/**
+ * Convert processed packet handle to event
+ *
+ * The packet handle must be an output of an compression/decompression operation.
+ *
+ * @param pkt      Packet handle from odp_comp_compress_enq()/
+ *                 odp_comp_decomp_enq()
+ *
+ * @return Event handle
+ */
+odp_event_t odp_comp_packet_to_event(odp_packet_t pkt);
```

Comment:
Could you please move `_from_event()`/`_to_event()` prototypes next to each other?